compiler/gradient_with_order.cc

```diff
@@ -132,7 +132,20 @@ void AddRetainedParts(Graph* fwd_graph, Graph* bwd_graph, std::map<Value*, Value
     }
 }
 
-void AddGradientNodesForTrainingWithOrders(Graph* fwd_graph, Graph* bwd_graph, const std::vector<Order>& orders) {
+bool IsComputationOrderSupported(const Graph& graph) {
+    for (const auto& value_ptr : graph.all_values()) {
```

Comment:
`for (const Value* value : graph.GetNecessaryValeus())` would be better since we don't care if unused value has dynamic shape?

Reply:
fixed.